ports/esp32/machine_uart.c

```diff
@@ -568,6 +587,15 @@ static const mp_irq_methods_t uart_irq_methods = {
 };
 
 static mp_irq_obj_t *mp_machine_uart_irq(machine_uart_obj_t *self, bool any_args, mp_arg_val_t *args) {
+    #if MICROPY_HW_ENABLE_UART_REPL
+    if (self->uart_num == MICROPY_HW_UART_REPL) {
+        mp_raise_ValueError(MP_ERROR_TEXT("UART does not support IRQs"));
```

Comment:
Why isn't this possible?  I would think IRQ still works on the UART REPL.

Reply:
Currently that's because of that we never initialize the UART queue for the REPL UART. See https://github.com/micropython/micropython/pull/8255. This error is a more specific case of the error raised below: Namely `self->uart_queue` will always be `NULL` for the REPL UART. For clarity I'll move it into the check below.

Reply:
OK, I see.  So user/Python IRQs on the UART REPL didn't work prior to this change anyway.